### DISCUSSION TITLE
Failing to add jpg files

### DISCUSSION THREAD

**Discussion by lucasHabkost** [photoprism/photoprism#2462] [Q&A]
Hi. I'm missing several photos after indexing an ~45k files library.

I tried reindexing specific folders and I see a lot of error of the type:
- index: failed cadding main jpg file 'FILENAMEXXX.jpg'
- index: failed creating thumbnails for 'FILENAMEXXX.jpg' (unexpected EOF)
- metadata: scan-data is unbonded; EOI not encountered before EOF while parsing jpeg file 'FILENAMEXXX.jpg'

The pictures were not manipulated in any shape or form, only moved between folders.

Any suggestions on how I can proceed?

Thanks!

**Comment by lastzero** [comment]
Seems related to this issue:
- https://github.com/golang/go/issues/45902

My understanding of the comments (you are welcome to read them as well) is that the JPEGs in question have small errors, such as missing data at the end. Essentially, the authors of the Go JPEG library decided to report an error rather than try to recover the image.

Note that there are tools that can repair the JPEGs, probably under the name jpeg fixer and similar.

**Reply by lastzero** [reply]
If the problem occurred after you moved files with WebDAV and they were not broken before, it could be a problem with our WebDAV server or the client you are using. If you are using other software or a network file system, that could also be a cause. 

**Should you suspect that the problem is related to any of the above, we will need information about the software versions and other information about your environment, such as browser, operating system, file system and webdav configuration, otherwise we will not be able to provide helpful advice:**

- https://photoprism.app/kb/reporting-bugs

**Reply by lastzero** [reply]
Try running `gm identify -verbose file.jpg` as recommended in the GitHub issue referenced above. You may first need to install this tool, depending on what operating system you use: http://www.graphicsmagick.org/identify.html

I'm also now calling Seek(0,0) before opening an image (on my local development version). No impact, the test file provided with the issue remains broken.

**Reply by lastzero** [reply]
When I run `convert eof.jpg eof-fixed.jpg`, the glitch is fixed i.e. the repaired copy `eof-fixed.jpg` can be opened without problems.

This command is part of ImageMagick:
```
 $ convert --help
Version: ImageMagick 6.9.10-23 Q16 x86_64 20190101 https://imagemagick.org
Copyright: © 1999-2019 ImageMagick Studio LLC
```

**Reply by lastzero** [reply]
I've documented the process here so we can eventually automate it:
- #2463

**Reply by lucasHabkost** [reply]
> Should you suspect that the problem is related to any of the above, we will need information about the software versions and other information about your environment, such as browser, operating system, file system and webdav configuration, otherwise we will not be able to provide helpful advice:

No, I have not used the WebDAV function.

I'll test the ImageMagick solution and see if that resolves and will circle back here!

**Reply by lucasHabkost** [reply]
Ok, so I tested the ImageMagick convert solution and it resolved the issue. I decided to convert "in-place", i.e. the "new file" ends up with the same name as the original file ($convert eof.jpg eof.jpg).

One thing worth mentioning is that perhaps an automated file fixing may incur on a pitall on the below scenario:
- I'm using Syncthing to sync the pictures from a laptop into my "Originals" folder on my HD plugged to Raspberry PI. 
- When I converted  the file, it created discrepancies between what is on the laptop folder and what is on my Originals folder, leading up to sync issues. 

I understand this is not a Photoprism generated issue, but it's additional information to take in consideration. :)

Thanks!

**Comment by 21m** [comment]
Just adding to the thread for anyone running into the same issue. I've being using WebDAV from nextcloud to sync with photoprism. If i upload files directly to photoprism no issues, if netxcloud then WebDAV, then i get the EOF issue.

I'll try the steps provided here and report back

**Comment by maxime1992** [comment]
Hello, has anyone found a good tool to fix images with EOF in batch? I've tried a few but nothing that really makes me want to stick with it. It seems to be with all the panoramas, and I've got a bunch of them :sweat_smile: ... Thanks!
